### PR TITLE
docs: Adding events to a plugin

### PR DESCRIPTION
**Pull request by swenson**
This adds a short doc describing the basic process of adding event notifications to a plugin as well
as some examples and best practices.

### CODE REVIEW

File: website/content/docs/plugins/plugin-development-event-notifications.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+# Event notifications development in Vault plugins
+
+<EnterpriseAlert product="vault" />
+
+~> Advanced topic! Plugin development is a highly advanced topic in Vault, and
+is not required knowledge for day-to-day usage. If you don't plan on writing any
+plugins, we recommend not reading this section of the documentation.
+
+[Event notifications](/vault/docs/concepts/events) can provide information when useful information is available from your plugin that might be useful to operators and clients of Vault clusters. Any plugin running in Vault Enterprise 1.16+ can generate such events.
```
Comment (fairclothjm) [inline website/content/docs/plugins/plugin-development-event-notifications.mdx:15]: nit: I might reword this slightly

```suggestion
[Event notifications](/vault/docs/concepts/events) can provide useful information from your plugin that might be helpful to operators and clients of Vault clusters. Any plugin running in Vault Enterprise 1.16+ can generate such events.
```

File: website/content/docs/plugins/plugin-development-event-notifications.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+This article assumes that the plugin is written in the Go programming language.
+It is possible for non-Go plugins to send events as well by calling the [`SendEvent` gRPC call](https://github.com/hashicorp/vault/blob/main/sdk/plugin/pb/backend.proto#L713) directly.
```
Comment (fairclothjm) [inline website/content/docs/plugins/plugin-development-event-notifications.mdx:18]: Should this be a permalink?

```suggestion
It is possible for non-Go plugins to send events as well by calling the [`SendEvent` gRPC call](https://github.com/hashicorp/vault/blob/da00addcb60e5fc4fef308715c0a1bd81c475167/sdk/plugin/pb/backend.proto#L713) directly.
```

File: website/content/docs/plugins/plugin-development-event-notifications.mdx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+While there are no strict rules for what events a plugin may generate, there are some best practices to follow:
+
+- **Do *not* include sensitive or secret data** in the event notification. Events are not protected by the same policy as the underlying secret.
+
+- Only generate event notificiations on writes and actionable errors, **not** reads. Generally, reads are more frequent than writes so generating event notifications for every read operation could have a negative impact on the server performance. Operators that need to be notified when information is read are usually better served by using Vault [audit logs](/vault/docs/audit).
+
+- Prefix event types with your plugin name to avoid clashing with other plugins' event types. For example, the database plugin uses the `database/` prefix for all of its event notifications.
+
+- Use [common metadata keys](https://github.com/hashicorp/vault/blob/main/sdk/logical/events.go#L13-L25), such as `data_path` for the API path to the secret the event notification is about, and `operation` to indicate what modification took place.
```
Comment (fairclothjm) [inline website/content/docs/plugins/plugin-development-event-notifications.mdx:50]:
```suggestion
- Use [common metadata keys](https://github.com/hashicorp/vault/blob/da00addcb60e5fc4fef308715c0a1bd81c475167/sdk/logical/events.go#L13-L25), such as `data_path` for the API path to the secret the event notification is about, and `operation` to indicate what modification took place.
```

**Review by fairclothjm — APPROVED** [review]
Looks great! Just a few suggestions :)

**Comment by swenson** [comment]
Thanks!

**Pull request merged**